### PR TITLE
[SPARK-11897][SQL] Add @scala.annotations.varargs to sql functions

### PR DESCRIPTION
**Pull request by xguo27**

### CODE REVIEW

**Comment by vanzin** [comment]
ok to test

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -689,6 +689,7 @@ object functions extends LegacyFunctions {
    * @group normal_funcs
    * @since 1.4.0
    */
+  @scala.annotation.varargs
   def array(colName: String, colNames: String*): Column = {
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: when we have this annotation, can we just use one parameter `colNames: String*`?

Comment (rxin) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: that will conflict with Column*, due to type erasure.

**Comment by rxin** [comment]
Thanks - I've merged this.

**Pull request closed without merging**